File: 1.0/src/c/cz/mg/vulkan/PFN_vkFreeCommandBuffers.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkFreeCommandBuffers_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong commandPool, jlong commandBufferCount, jlong pCommandBuffers){
    (void)env;
    (void)jc;
    PFN_vkFreeCommandBuffers f = (PFN_vkFreeCommandBuffers)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkCommandPool*)jniLongToPointer(commandPool)),
        *((uint32_t*)jniLongToPointer(commandBufferCount)),
        ((VkCommandBuffer*)jniLongToPointer(pCommandBuffers))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkFreeCommandBuffers.java
package cz.mg.vulkan;

public class PFNvkFreeCommandBuffers extends VkFunctionPointer {
    public PFNvkFreeCommandBuffers() {
    }

    protected PFNvkFreeCommandBuffers(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkFreeCommandBuffers(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkFreeCommandBuffers(long value) {
        setValue(value);
    }

    public PFNvkFreeCommandBuffers(VkInstance instance) {
        super(instance, new VkString("vkFreeCommandBuffers"));
    }

    public void call(VkDevice device, VkCommandPool commandPool, int commandBufferCount, VkCommandBuffer pCommandBuffers){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), commandPool != null ? commandPool.getVkAddress() : VkPointer.getNullAddressNative(), commandBufferCount, pCommandBuffers != null ? pCommandBuffers.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long device, long commandPool, int commandBufferCount, long pCommandBuffers);

}
